cpp/src/arrow/engine/substrait/extension_set.cc

```diff
@@ -858,11 +867,34 @@ struct DefaultExtensionIdRegistry : ExtensionIdRegistryImpl {
 
     // -------------- Substrait -> Arrow Functions -----------------
     // Mappings with a _checked variant
-    for (const auto& function_name : {"add", "subtract", "multiply", "divide"}) {
+    for (const auto& function_name :
+         {"add", "subtract", "multiply", "divide", "power", "sqrt", "abs"}) {
```

Comment:
`sqrt` isn't currently defined as overflowable in Substrait.

Reply:
Ha -- good catch. I changed the `DecodeOptionlessUncheckedArithmetic` and updated the tests.